app/controllers/products_controller.rb

```diff
@@ -8,13 +8,12 @@ class ProductsController < ApplicationController
   before_action :load_evaluates, only: :show
 
   def index
-    product = Product.includes(:category)
-    @products = product.filter_by(params[:name]).paginate page: params[:page],
+    @search = Product.includes(:category).ransack params[:q]
```

Comment:
params[:q] anh đổi tên param được k em

Reply:
@search là object của class nào vây jem

Reply:
1. Dạ đổi tên param[:q] vẫn được anh, nhưng phải config search_key trong file config/initializers/ransack.rb. Mà lệnh để tạo ra file config này thì em chưa rõ  :))
2. Khi mình điền sai object này cho method search_form_for thì Rails nó báo cái lỗi đây 'No Ransack::Search object was provided to search_form_for' nên em nghĩ @search ở đây là đối tượng của class Search trong module Ransack anh :))